### ISSUE TITLE
Optimize storage and routing algorithm for memory restricted environments like Android

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#1]
The routing algorithms can already be used on Android - see https://github.com/graphhopper/graphhopper/wiki/Android

But there should be a graph implementation which is disc based to avoid loading the full graph into memory. Otherwise only small road networks are possible.

Additionally there should be a memory bound A\* implementation like http://en.wikipedia.org/wiki/SMA* to reduce memory consumption while executing the algorithm. Or CH could help here too.

**Comment by karussell** [comment]
For none-CH requests one can now increase the heuristical nature and improve speed by a factor of at least 2. See #506

**Comment by easbar** [comment]
Almost eight years later issue #1 got closed :smile:

**Comment by karussell** [comment]
WTH :D

**Comment by easbar** [comment]
Uh yeah there was a commit in the navigation repo that closed issue number 1 of the **navigation** repository:

![image](https://user-images.githubusercontent.com/17603532/87218221-06149180-c351-11ea-8d2a-74053cbdcacc.png)

and now we had this commit here it closed issue number 1 of **this** repository. Btw were there other issue references (in comments etc.) pointing to navigation issues? For map matching we should also check this and maybe change comments like 

```java
// this is a comment, see #6586
``` 

to 

```java
// this is a comment, see map-matching/#6586
```

**Comment by boldtrn** [comment]
Oh no, that's really unfortunate. I guess there will be multiple comments like this in the MapMatching Repo?

**Comment by easbar** [comment]
Yes, but in the navigation code there weren't any. For map-matching we should not forget about this. But actually we should be a bit careful because I think we decided not to squash the map-matching commits (to preserve history) and there might be some commits that auto-fix some issues :)

**Comment by karussell** [comment]
> maybe change comments like

We cannot change commit messages without changing the history that could break other things.

**Comment by easbar** [comment]
> We cannot change commit messages without changing the history that could break other things.

I meant we should change comments in the code. But even if there is something like `fixes #256` in a commit message we could change the commit message when we are merging map-matching? It would not be possible in the map-matching repository, but when we merge it here its no problem to change the commits (they are just new commits for this repository, nothing existing is changed)?

**Comment by karussell** [comment]
I do not like rewriting history for map matching that much as e.g. we would not be able to easily find map matching commits in the graphhopper repo and maybe other disadvantages. E.g. when linking to commits like [here](https://github.com/graphhopper/map-matching/issues/40#issuecomment-161539479 although this is rather rare (found 4 issues).

**Comment by easbar** [comment]
Ok sure its probably also not a real problem. If some issues are auto-closed we can just re-open them. But do the commit hashes really stay the same even when porting the map-matching repo to here?

**Comment by karussell** [comment]
Yes. Or at least they did it with the method I used for the navigation pull request. See e.g. the first commit here: https://github.com/graphhopper/graphhopper/pull/2071/commits -> 6c03fdd34e8e71fee4238ac4ac8edb3faa4bc54b -> https://github.com/graphhopper/graphhopper-navigation/commit/6c03fdd34e8e71fee4238ac4ac8edb3faa4bc54b
